packages/core/src/project-creation.ts

```diff
@@ -0,0 +1,145 @@
+import path from 'node:path'
+import fs from 'node:fs'
+import os from 'node:os'
+import { getRecommendedGitIgnore } from './project-structure'
+import { getPackageRoot } from './util/packageInfo'
+import { prompt } from 'enquirer'
+
+function createConfirmationPrompt(name: string, message: string) {
+  return {
+    type: 'confirm',
+    name,
+    message,
+    initial: 'y',
+    default: '(Y/n)',
+    isTrue(input: string | boolean) {
+      if (typeof input === 'string') {
+        return input.toLowerCase() === 'y'
+      }
+
+      return input
+    },
+    isFalse(input: string | boolean) {
+      if (typeof input === 'string') {
+        return input.toLowerCase() === 'n'
+      }
+
+      return input
+    },
+    format(): string {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const that = this as any
+      const value = that.value === true ? 'y' : 'n'
+
+      if (that.state.submitted === true) {
+        return that.styles.submitted(value)
+      }
+
+      return value
+    },
+  }
+}
+
+export async function confirmProjectCreation(): Promise<{
+  projectRoot: string
+  shouldAddGitIgnore: boolean
+}> {
+  return prompt([
+    {
+      name: 'projectRoot',
+      type: 'input',
+      initial: process.cwd(),
+      message: 'Kuai project root:',
+    },
+    createConfirmationPrompt('shouldAddGitIgnore', 'Do you want to add a .gitignore?'),
+  ])
+}
+
+async function copySampleProject(projectRoot: string) {
+  const packageRoot = getPackageRoot()
+
+  fs.existsSync(projectRoot)
+
+  const sampleProjectPath = path.join(packageRoot, 'sample-projects')
+
+  const sampleProjectRootFiles = fs.readdirSync(sampleProjectPath)
+  const existingFiles = sampleProjectRootFiles
+    .map((f) => path.join(projectRoot, f))
+    .filter((f) => fs.existsSync(f))
+    .map((f) => path.relative(process.cwd(), f))
+
+  if (existingFiles.length > 0) {
+    const errorMsg = `We couldn't initialize the sample project because these files already exist: ${existingFiles.join(
+      ', ',
+    )}
+
+Please delete or move them and try again.`
+    console.log(console.log(errorMsg))
+    process.exit(1)
+  }
+
+  fs.cpSync(path.join(packageRoot, 'sample-projects'), projectRoot, { recursive: true })
+}
+
+async function addGitIgnore(projectRoot: string) {
+  const gitIgnorePath = path.join(projectRoot, '.gitignore')
+
+  let content = await getRecommendedGitIgnore()
+
+  if (fs.existsSync(gitIgnorePath)) {
+    const existingContent = fs.readFileSync(gitIgnorePath, 'utf-8')
+    content = `${existingContent}
+${content}`
```

Comment:
`fs.appendFileSync` should work here

Ref: [fs.appendFileSync](https://nodejs.org/api/fs.html#fsappendfilesyncpath-data-options)